book/src/porting/implementing_sysdeps.md

```diff
@@ -0,0 +1,153 @@
+# Implementing your sysdeps
+
+## Telling mlibc about your sysdeps
+
+In mlibc's top level `meson.build` file, you'll see a large `if`/`else` chain that selects the right sysdeps subdirectory based on the system name. We'll add ourselves here:
+```meson
+{{#include ../../../meson.build:demo-sysdeps}}
+    # ...
+```
+
+Now, create the `sysdeps/demo` folder. At minimum it should contain a `meson.build` file where we'll declare all the `.cpp` files and select header files to install:
+
+```meson
+{{#include ../../../sysdeps/demo/meson.build:sources-and-includes}}
+```
+
+The `sysdep_supported_options` tells mlibc which 'options' your sysdeps supports. For example, the `unistd.h` header is only available when the POSIX option is enabled.
+
+## Preparing to build mlibc
+
+Now, from a fresh clone of `mlibc` you can run:
+
+```
+$ meson setup --cross-file=path/to/your.cross-file -Ddefault_library=static -Dbuildtype=debug build
+```
+
+The `-Ddefault_library=static` tells meson to only produce a statically linked library (`libc.a`). We *strongly* recommend getting statically linked binaries to work before dynamically linked ones.
```

Comment:
Not sure if we should strongly recommend (instead of just suggesting) static linkage first, it's not that hard to go to dynamic linking anyways.